gamemodes/terrortown/entities/weapons/weapon_tttbasegrenade.lua

```diff
@@ -84,6 +85,123 @@ function SWEP:SecondaryAttack()
 
 end
 
+---
+-- @param Vector currentIterationPosition The initial position for this iteration cycle.
+-- @param number launchVelocity The launch velocity for this iteration cycle.
+-- @param number gravityMultiplier The gravity multiplier for this iteration cycle.
+-- @param number stepDistance The step size for the current iteration cycle.
+-- @return Vector The next location to iterate from.
+local function PositionFromPhysicsParams(currentIterationPosition, launchVelocity, gravityMultiplier, stepDistance)
+	local activeGravity = gravityMultiplier * physenv.GetGravity()
+
+	return currentIterationPosition + (launchVelocity * stepDistance + 0.5 * activeGravity * stepDistance ^ 2)
+end
+
+if CLIENT then
+	---
+	-- @realm client
+	local cvEnableTrajectoryUI = CreateConVar("ttt2_grenade_trajectory_ui", 0, FCVAR_ARCHIVE)
+
+	local function AlphaLerp(from, frac, max)
+		local fr = frac ^ 0.5
+		return ColorAlpha(from, Lerp(fr, 0, math.min(max, 255)))
+	end
+
+	---
+	-- @param Player ply
+	-- @realm client
+	function SWEP:DrawDefaultThrowPath(ply)
+		local stepSize = 0.005
+		local gravityMultiplier = 1
+
+		local owner = self:GetOwner()
+		local currentIterationPosition, _ = self:GetViewModelPosition(owner:EyePos(), owner:EyeAngles())
+		local launchPosition, launchVelocity = self:GetThrowVelocity()
+		currentIterationPosition = currentIterationPosition - ply:EyePos() + launchPosition
+		local previousIterationPosition = PositionFromPhysicsParams(currentIterationPosition, launchVelocity, gravityMultiplier, stepSize)
+
+		--
+		local fractionalFrameTimeForDashedLineSegmentLengthPurposes = (SysTime() % 1) * 2
+		local evenOrOddFrameForDashedLineDrawingPurposes = fractionalFrameTimeForDashedLineSegmentLengthPurposes > 1 and 1 or 0
+		fractionalFrameTimeForDashedLineSegmentLengthPurposes = fractionalFrameTimeForDashedLineSegmentLengthPurposes - math.floor(fractionalFrameTimeForDashedLineSegmentLengthPurposes)
```

Comment:
you know these variable names are ridiculous

Reply:
even so, i don't know any middleground between what they were and what they are

Reply:
Dunno, maybe something like 'fractionalFrameTime'? It doesn't have to describe it in detail, it just shouldn't be called `a`

Reply:
ok

Reply:
the line below as well, then it is fine with me